### PR TITLE
Load StripeGateway from the container

### PR DESCRIPTION
**Pull request by phleobon**
Loading the Laravel\Cashier\StripeGateway from the containers allows to mock it in functional test.

While trying to test the user registration + subscription process, you might not want to hit the stripe API.

### CODE REVIEW

File: src/Laravel/Cashier/Billable.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ public function charge($amount, array $options = [])
      */
     public function subscription($plan = null)
     {
-        return new StripeGateway($this, $plan);
+        return app('Laravel\Cashier\StripeGateway', [$this, $plan]);
```
Comment (GrahamCampbell) [inline src/Laravel/Cashier/Billable.php]: You can't use the app helper because it's not available in non-laravel app.s

Comment (GrahamCampbell) [inline src/Laravel/Cashier/Billable.php]: Cashier does not depend on laravel.

**Comment by phleobon** [comment]
Oh, I wasn't aware about this.

**Comment by phleobon** [comment]
By the way, any advice on how I could mock it then ?

**Comment by GrahamCampbell** [comment]
Instance mocking perhapse?

**Comment by phleobon** [comment]
I'll try that. Thanks

**Pull request closed without merging**